src/serde/deserialize_program.rs

```diff
@@ -167,7 +167,27 @@ where
     let n = Number::deserialize(deserializer)?;
     match Felt252::parse_bytes(n.to_string().as_bytes(), 10) {
         Some(x) => Ok(Some(x)),
-        None => Err(String::from("felt_from_number parse error")).map_err(de::Error::custom),
+        None => {
+            // Handle de Number with scientific notation cases
+            // e.g.: Number(1e27)
+            let str = n.to_string();
+            let list: [&str; 2] = str
+                .split('e')
+                .collect::<Vec<&str>>()
+                .try_into()
+                .map_err(|_| String::from("felt_from_number parse error"))
+                .map_err(de::Error::custom)?;
+
+            let base = Felt252::parse_bytes(list[0].to_string().as_bytes(), 10).ok_or(
+                de::Error::custom(String::from("felt_from_number parse error")),
+            )?;
+            let exponent = list[1]
+                .parse::<u32>()
+                .map_err(|_| de::Error::custom(String::from("felt_from_number parse error")))?;
```

Comment:
Maybe we could wrap this logic in a function that returns an `Option` and then map that option to a Result here as to avoid declaring the same error so many times

Reply:
Nice Idea!
Done [98b60bc](https://github.com/lambdaclass/cairo-rs/pull/1188/commits/98b60bcb3bff807cb46ee7c3ae60dfe3032ccf9c)